### PR TITLE
Create weather data/iotools page in User's Guide

### PR DESCRIPTION
**Pull request by kandersolar**
- ~[ ] Closes #xxxx~
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - ~[ ] Tests added~
 - ~[ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - ~[ ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

This PR creates a new page in the User's Guide focused on weather data and `pvlib.iotools`.  It's rather bare bones at the moment.  

See also https://github.com/pvlib/pvlib-python/discussions/1304

### CODE REVIEW

**Comment by AdamRJensen** [comment]
It seems that most people need weather data from scratch, so I would propose a restructured outline along the lines of:

- **Weather data sources** - discussing free options in terms of _satellite data_ (NSRDB, PVGIS, CAMS) and _measurements_ (BSRN, SRML, MIDC, SURFRAD, SOLRAD).
- **File readers** - discussing TMY2, TMY3, and EPW
- **Usage** - describe the general iotools convention

My main objective for this page is to be able to have a link to send someone every time we get the questions "_I need weather data, what do I do?_"

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
+.. _weatherdata:
+
+Importing weather data
+======================
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:4]:
```suggestion
Weather data
=========
```

Comment (AdamRJensen) [reply docs/sphinx/source/user_guide/weather_data.rst:4]: I think it's more than just "importing", it's also obtaining/retrieving weather data.

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Many PV modeling applications require irradiance and meteorological datasets
+as the inputs to a PV system model.  These weather datasets are available
+from many sources and in many formats.  The :py:mod:`pvlib.iotools` module
+contains functions to retrieve and import these datasets in a form
+that is convenient to use with the rest of pvlib.
+
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:11]:
```suggestion
Simulating the performance of a PV system requires irradiance and meteorological data
as the inputs to a PV system model.  Weather datasets are available
from many sources and in many formats.  The :py:mod:`pvlib.iotools` module
contains functions to easily retrieve and import such datasets in a standardized
form that is convenient to use with the rest of pvlib.

```

**Comment by harry-solcast** [comment]
Hi, I'd like to suggest putting in a category to help explain what kind of sources of weather data are available.

Disclaimer: I work for Solcast, and we use Pvlib internally, and many of our customers use it too. I hope this suggestion comes across as helping users to understand the types of data available.

```
Other weather sources
-------------------------
Measurement Data Sources
From in-situ monitoring equipment. If properly maintained and quality-controlled, these are the highest quality source of weather information. However, the coverage depends on a weather station having been set up in advance for the location and times of interest. There are datasets such as BSRN and SURFRAD which make their measurement data publicly available.


Numerical Weather Prediction (NWP) Data Sources
These are mathematical simulations of weather systems. The data quality is much lower than that of measurements, owing in part to coarser spatial and temporal resolution, as well as many models not being optimised for solar irradiance for PV applications. On the plus side, these models typically have worldwide coverage, with some regional models (e.g. HRRR) sacrifice global coverage for somewhat higher spatial and temporal resolution. Various forecast (e.g. GFS, ECMWF, ICON) and reanalysis sources (ERA5, MERRA2) exist.


Satellite Data Sources
These sources process satellite imagery (typically from geostationary satellites) to identify and classify clouds, and combine this with solar irradiance models to produce irradiance estimates. The quality is generally much higher than NWP, but still not as good as a well-maintained weather station. They have high spatial and temporal resolution corresponding to the source satellite imagery, and are generally optimised to estimate solar irradiance for PV applications. Free sources such as PVGIS are available, and commerical sources such as SolarAnywhere, Solcast and Solargis provide paid options though often have free trials.
```

**Comment by kandersolar** [comment]
Thanks @harry-solcast!  Very helpful :)

I suppose this PR is ready for review.  Here's the rendered version: https://pvlib-python--1754.org.readthedocs.build/en/1754/user_guide/weather_data.html

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 3]
```diff
@@ -0,0 +1,159 @@
+.. _weatherdata:
+
+Weather data
+============
+
+Simulating the performance of a PV system requires irradiance and meteorological data
+as the inputs to a PV system model.  Weather datasets are available
+from many sources and in many formats.  The :py:mod:`pvlib.iotools` module
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:8]: Could we make a hyperlink to the iotools module for those who do not know what it is?

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+contains functions to easily retrieve and import such datasets in a standardized
+form that is convenient to use with the rest of pvlib.  
+
+The primary focus of :py:mod:`pvlib.iotools` is time series solar resource
+data like the datasets from PVGIS and the NSRDB, but it also provides
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:13]:
```suggestion
data like the irradiance datasets from PVGIS and the NSRDB, but it also provides
```

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+functionality for other types of data useful for certain aspects of PV modeling
+(e.g. precipitation data from :py:func:`~pvlib.iotools.get_acis_prism`
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:15]:
```suggestion
functionality for other types of data useful for certain aspects of PV modeling.
For example, precipitation data is available from :py:func:`~pvlib.iotools.get_acis_prism`
```

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+overview of different types of weather data, and when you might want to use
+them.
+
+1. **Ground station measurements**:
+   From in-situ monitoring equipment. If properly maintained and
+   quality-controlled, these are the highest quality
+   source of weather information. However, the coverage depends on
+   a weather station having been set up in advance for the location and
+   times of interest. Some ground station networks like the BSRN and SURFRAD
+   make their measurement data publicly available.
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:37]:
```suggestion
   make their measurement data publicly available. An global overview of ground
   stations is available at https://SolarStations.org.
```

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+   
+   Data from public ground station measurement networks are useful if you
+   want accurate, high-resolution data but have flexibility around the
+   specific measurement location.
+
+2. **Satellite data**: 
+   These sources process satellite imagery (typically from geostationary
+   satellites) to identify and classify clouds, and combine this with solar
+   irradiance models to produce irradiance estimates. The quality is
+   generally much higher than NWP, but still not as good as a well-maintained
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:47]:
```suggestion
   irradiance models and aerosol data to produce irradiance estimates. The
   quality is generally much higher than NWP, but still not as good as a well-maintained
```

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+   weather station. They have high spatial and temporal resolution
+   corresponding to the source satellite imagery, and are generally
+   optimised to estimate solar irradiance for PV applications. Free sources
+   such as PVGIS are available, and commerical sources such as SolarAnywhere,
+   Solcast and Solargis provide paid options though often have free trials.
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:52]:
```suggestion
   such as PVGIS, NSRDB, and CAMS are available, and commerical sources such
   as SolarAnywhere, Solcast, and Solargis provide paid options though often
   have free trials.
```

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+   
+   Satellite data is useful when suitable ground station measurements are
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:54]:
```suggestion
   Satellite data is useful when suitable ground measurements are
```

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+   resolution, as well as many models not being optimised for solar
+   irradiance for PV applications. On the plus side, these models typically
+   have worldwide coverage, with some regional models (e.g. HRRR) sacrifice
+   global coverage for somewhat higher spatial and temporal resolution.
+   Various forecast (e.g. GFS, ECMWF, ICON) and reanalysis sources (ERA5,
+   MERRA2) exist.
+   
+   NWP datasets are primarily useful for parts of the world not covered
+   by satellite-based datasets (e.g. the poles) or if extremely long time
+   ranges are needed.
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:70]:
```suggestion
   by satellite-based datasets (e.g. polar regions) or if extremely long time
   ranges are needed.
```

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+
+For a more detailed comparison of the weather datasets available through
+pvlib, see [1]_.
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:73]: Can we move this citation up to the intro (I doubt most folks would read this far)

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+
+
+:py:mod:`pvlib.iotools` usage
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:76]:
```suggestion
Usage
```

File: docs/sphinx/source/user_guide/weather_data.rst [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+:py:func:`~pvlib.iotools.get_psm3`, :py:func:`~pvlib.iotools.get_pvgis_tmy`,
+and so on.
+
+For satellite/reanalysis datasets, the location is specified by latitude and
+longitude in decimal degrees:
+
+.. code-block:: python
+
+    lat, lon = 33.75, -84.39  # Atlanta, Georgia, United States
+    df, metadata = pvlib.iotools.get_psm3(lat, lon, map_variables=True, ...)
```
Comment (AdamRJensen) [inline docs/sphinx/source/user_guide/weather_data.rst:125]:
```suggestion
    latitude, longitude = 33.75, -84.39  # Atlanta, Georgia, United States
    df, metadata = pvlib.iotools.get_psm3(latitude, longitude, map_variables=True, ...)
```

**Review by AdamRJensen — APPROVED** [review]
This is awesome. What a great contribution!

**Comment by AdamRJensen** [comment]
Thank you @kandersolar !

**Pull request merged**